### ISSUE TITLE
[Android] Crash when canceled `showScreenSet()`

### ISSUE THREAD

**Issue by hiroki-murakami-AT-IRET** [SAP/gigya-flutter-plugin#37]
On Android devices, when back from Web Screen-Set with hardware back key, `EventChannel('screensetEvents')` throws the following error (uncatchable) and the application crashes.

```
PlatformException(200001, Operation canceled, null, null)
```

The cause of this is the flutter side implementation of the plugin do not pass `onError` to `EventChannel('screensetEvents').receiveBroadcastStream().listen()`.

```
    const EventChannel _stream = EventChannel('screensetEvents');
    _screenSetsEventStream = _stream.receiveBroadcastStream().listen((onData) {
      onScreenSetEvent(
        onData['event'],
        onData['data'],
      );
      if (onData['event'] == 'onHide' || onData['event'] == 'onCanceled') {
        _screenSetsEventStream = null;
      }
    });
```

But, I think, Android's `onCanceled` should be returned  by `success()` instead of `error()` like `onError` in the first place.

Not really good:
```
            override fun onCanceled() {
                screenSetsEventsSink?.error("200001", "Operation canceled", null)
                screenSetsEventsHandler = null
                screenSetEventsChannel = null
                screenSetsEventsSink = null
            }
```

LGTM:
```
            override fun onCanceled() {
                screenSetsEventsSink?.success(
                    mapOf(
                        "event" to "onCanceled",
                        "data" to mapOf<String, Any>()
                    )
                )
                screenSetsEventsHandler = null
                screenSetEventsChannel = null
                screenSetsEventsSink = null
            }
```

Thank you.

**Comment by navaronbracke** [comment]
@hiroki-murakami-AT-IRET I have a WIP PR which refactors a bunch of things in the plugin. One of which is how the screen sets stream is handled. Your proposed change (to have all events be regular data events) is included in that PR.

**Comment by hiroki-murakami-AT-IRET** [comment]
Fixed in 1.0.0 release.
Thank you!
